File: app/src/main/java/org/ametro/ui/activities/SettingsList.java
package org.ametro.ui.activities;

import android.os.Bundle;
import androidx.core.app.NavUtils;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.MenuItem;

import org.ametro.R;

public class SettingsList extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings_list_view);

        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
        ActionBar actionBar = getSupportActionBar();
        if(actionBar!=null) {
            actionBar.setDefaultDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                NavUtils.navigateUpFromSameTask(this);
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/org/ametro/ui/activities/SettingsList.kt
package org.ametro.ui.activities

import android.annotation.SuppressLint
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.widget.Toolbar
import org.ametro.R
import androidx.core.app.NavUtils
import org.ametro.app.Constants
import org.ametro.databinding.ActivitySettingsListViewBinding
import org.ametro.ui.fragments.SettingsListFragment
import org.ametro.utils.misc.UIUtils

class SettingsList : AppCompatActivityEx() {
    private lateinit var binding: ActivitySettingsListViewBinding

    @SuppressLint("RestrictedApi")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySettingsListViewBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.includeToolbar.toolbar)

        supportActionBar?.run {
            setDefaultDisplayHomeAsUpEnabled(true)
            setDisplayHomeAsUpEnabled(true)
        }

        if (Build.VERSION.SDK_INT >= Constants.INSETS_MIN_API) {
            setFitSystemWindowsFlags(binding.root, keepNavbar = true)
            setNavbarSolid()
            val toolbar = binding.includeToolbar.toolbar
            val toolbarApplier = UIUtils.makeTopInsetsApplier(toolbar)
            toolbar.setOnApplyWindowInsetsListener { _, insets ->
                toolbarApplier.applyInset(insets)
                insets
            }
        }

        supportFragmentManager
            .beginTransaction()
            .replace(R.id.list, SettingsListFragment())
            .commit()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                NavUtils.navigateUpFromSameTask(this)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}